apps/site/assets/ts/hooks/__tests__/useAlertsTest.tsx

```diff
@@ -32,10 +54,46 @@ describe("useAlertsByStop", () => {
     const { result, waitFor } = renderHook(() => useAlertsByStop("stop-id"), {
       wrapper: HookWrapper
     });
-    await waitFor(() => expect(result.current).toEqual(testAlert));
+    await waitFor(() => expect(result.current).toEqual([testAlert]));
   });
 
   afterAll(() => {
     global.fetch = unmockedFetch;
   });
 });
+
+describe("useAlertsByRoute", () => {
+  beforeAll(() => {
+    // provide mocked network response
+    global.fetch = jest.fn(url => {
+      if (url === "/api/alerts?route_ids=route-id") {
+        return oneAlertPromise;
+      } else if (url === "/api/alerts?route_ids=route-1,route-2") {
+        return twoAlertsPromise;
+      } else {
+        throw Error(
+          `Unexpected URL: ${url}.  The URL parsing parsed route ids into an unexpected format`
+        );
```

Comment:
feedback wanted:  I was struggling to get this error to surface if an unexpected url is passed to this mock function.  Right know if an invalid URL is received the test fails with a timeout instead of surfacing this as the reason.  Would be nice to find a way error that the URL parsing behaved in an unexpected way

Reply:
Huh, I wonder if it it because the error isn't thrown within a promise?
I've previously used [this resource](https://jestjs.io/docs/asynchronous#asyncawait) for testing async code; that may be of help here!